### ISSUE TITLE
autosubmit status

### ISSUE THREAD

**Issue by kinow** [BSC-ES/autosubmit#281]
In GitLab by @pablohe on Oct 26, 2017, 13:08

To get in get the current status of the jobs the user call `autosubmit monitor YYYY` and autosubmit produce a plot.

It is possible add an option that provide the status of the jobs in text mode, something like (or/and a xml file)?:

```
> autosubmit status YYYY

t00o_LOCAL_SETUP COMPLETED
t00o_REMOTE_SETUP COMPLETED
t00o_19900101_fc0_INI COMPLETED
t00o_19900101_fc0_1_SIM COMPLETED
t00o_19900101_fc0_1_POST COMPLETED
t00o_19900101_fc0_2_SIM FAILED
t00o_19900101_fc0_1_CLEAN WAITING
t00o_19900101_fc0_2_TRANSFER WAITING
t00o_19900101_fc0_2_LOCALPOST WAITING
t00o_19900101_fc0_2_PLOT WAITING
t00o_19900101_fc0_2_POST WAITING
t00o_19900101_fc0_2_CLEAN WAITING
```

**Comment by kinow** [comment]
In GitLab by @pablohe on Nov 23, 2017, 10:37

Yes, I think sould work.

**Comment by kinow** [comment]
In GitLab by @ lbatista on Nov 23, 2017, 11:53

created branch [`281-autosubmit-status`](https://earth.bsc.es/gitlab/es/autosubmit/compare/master...281-autosubmit-status)

**Comment by kinow** [comment]
In GitLab by @ lbatista on Nov 23, 2017, 11:56

closed

**Comment by kinow** [comment]
In GitLab by @pablohe on Nov 23, 2017, 15:50

reopened

**Comment by kinow** [comment]
In GitLab by @pablohe on Nov 23, 2017, 16:03

Thanks @ lbatista ! wonderful

Could you extend it a bit more? Will be nice if you add an option that provide the status and the log in txt mode and not produce the plot. ie:

```
> autosubmit monitor --txt

```

and the log could be something like:

```
...
t00o_19900101_fc0_1_POST COMPLETED /esnas/autosubmit/t00o/tmp/LOG_t00o/t00o_19900101_fc0_1_POST.20171123031859.err /esnas/autosubmit/t00o/tmp/LOG_t00o/t00o_19900101_fc0_1_POST.20171123031859.out
t00o_19900101_fc0_2_SIM FAILED /esnas/autosubmit/t00o/tmp/LOG_t00o/t00o_19900101_fc0_2_SIM.20171123031736.err /esnas/autosubmit/t00o/tmp/LOG_t00o/t00o_19900101_fc0_2_SIM.20171123031736.out
t00o_19900101_fc0_1_CLEAN WAITING /esnas/autosubmit/t00o/tmp/LOG_t00o/t00o_19900101_fc0_1_CLEAN.20171123032809.err /esnas/autosubmit/t00o/tmp/LOG_t00o/t00o_19900101_fc0_1_CLEAN.20171123032809.out
...

```

**Comment by kinow** [comment]
In GitLab by @ lbatista on Jan 29, 2018, 16:59

closed

**Comment by kinow** [comment]
In GitLab by @pablohe on Jan 29, 2018, 17:26

Theare some empty path when the job is waiting
```
 cat /esnas/autosubmit/t027/status/t027_20180129_1718.txt
t027_LOCAL_SETUP COMPLETED /esnas/autosubmit/t027/tmp/LOG_t027/t027_LOCAL_SETUP.20180111174930.out /esnas/autosubmit/t027/tmp/LOG_t027/t027_LOCAL_SETUP.20180111174930.err
t027_REMOTE_SETUP COMPLETED /esnas/autosubmit/t027/tmp/LOG_t027/t027_REMOTE_SETUP.20180111175128.out /esnas/autosubmit/t027/tmp/LOG_t027/t027_REMOTE_SETUP.20180111175128.err
t027_19900101_fc0_INI SUSPENDED /esnas/autosubmit/t027/tmp/LOG_t027/t027_19900101_fc0_INI.20180111174930.out /esnas/autosubmit/t027/tmp/LOG_t027/t027_19900101_fc0_INI.20180111174930.err
t027_19900101_fc0_1_SIM WAITING /esnas/autosubmit/t027/tmp/LOG_t027/ /esnas/autosubmit/t027/tmp/LOG_t027/
t027_19900101_fc0_2_SIM WAITING /esnas/autosubmit/t027/tmp/LOG_t027/ /esnas/autosubmit/t027/tmp/LOG_t027/
t027_19900101_fc0_1_POST WAITING /esnas/autosubmit/t027/tmp/LOG_t027/t027_19900101_fc0_1_POST.20180111174930.out /esnas/autosubmit/t027/tmp/LOG_t027/t027_19900101_fc0_1_POST.20180111174930.err
t027_19900101_fc0_2_POST WAITING /esnas/autosubmit/t027/tmp/LOG_t027/ /esnas/autosubmit/t027/tmp/LOG_t027/
t027_19900101_fc0_1_CMOROCE WAITING /esnas/autosubmit/t027/tmp/LOG_t027/t027_19900101_fc0_1_CMOROCE.20180111174930.out /esnas/autosubmit/t027/tmp/LOG_t027/t027_19900101_fc0_1_CMOROCE.20180111174930.err
t027_19900101_fc0_2_CMOROCE WAITING /esnas/autosubmit/t027/tmp/LOG_t027/ /esnas/autosubmit/t027/tmp/LOG_t027/
t027_19900101_fc0_1_CMORATM1 WAITING /esnas/autosubmit/t027/tmp/LOG_t027/t027_19900101_fc0_1_CMORATM1.20180111174930.out /esnas/autosubmit/t027/tmp/LOG_t027/t027_19900101_fc0_1_CMORATM1.20180111174930.err
t027_19900101_fc0_2_CMORATM1 WAITING /esnas/autosubmit/t027/tmp/LOG_t027/ /esnas/autosubmit/t027/tmp/LOG_t027/
t027_19900101_fc0_1_CLEAN WAITING /esnas/autosubmit/t027/tmp/LOG_t027/t027_19900101_fc0_1_CLEAN.20180111174930.out /esnas/autosubmit/t027/tmp/LOG_t027/t027_19900101_fc0_1_CLEAN.20180111174930.err
t027_19900101_fc0_2_CLEAN WAITING /esnas/autosubmit/t027/tmp/LOG_t027/ /esnas/autosubmit/t027/tmp/LOG_t027/
t027_19900101_fc0_CLEAN_MEMBER WAITING /esnas/autosubmit/t027/tmp/LOG_t027/t027_19900101_fc0_CLEAN_MEMBER.20180111174930.out /esnas/autosubmit/t027/tmp/LOG_t027/t027_19900101_fc0_CLEAN_MEMBER.20180111174930.err
t027_19900101_fc0_2_TRANSFER WAITING /esnas/autosubmit/t027/tmp/LOG_t027/t027_19900101_fc0_2_TRANSFER.20180111174930.out /esnas/autosubmit/t027/tmp/LOG_t027/t027_19900101_fc0_2_TRANSFER.20180111174930.err
t027_19900101_fc0_2_LOCALPOST WAITING /esnas/autosubmit/t027/tmp/LOG_t027/t027_19900101_fc0_2_LOCALPOST.20180111174930.out /esnas/autosubmit/t027/tmp/LOG_t027/t027_19900101_fc0_2_LOCALPOST.20180111174930.err
t027_19900101_fc0_2_PLOT WAITING /esnas/autosubmit/t027/tmp/LOG_t027/t027_19900101_fc0_2_PLOT.20180111174930.out /esnas/autosubmit/t027/tmp/LOG_t027/t027_19900101_fc0_2_PLOT.20180111174930.err
```

Also now I realize the job log is useful after it ran. So for COMPLETED and FAILED status.

thanks

**Comment by kinow** [comment]
In GitLab by @pablohe on Jan 29, 2018, 17:26

reopened

**Comment by kinow** [comment]
In GitLab by @ lbatista on Jan 29, 2018, 17:28

Hi @pablohe I can add the constraint to output the log paths only for COMPLETED and FAILED jobs. In that case, the empty path problem for WAITING jobs would also be solved

**Comment by kinow** [comment]
In GitLab by @pablohe on Jan 29, 2018, 18:14

Win win solution :) great!

**Comment by kinow** [comment]
In GitLab by @ lbatista on Feb 1, 2018, 20:02

closed

**Comment by kinow** [comment]
In GitLab by @pablohe on Feb 7, 2018, 16:21

in the failed jobs, no log is show

```bash
...
WS/shared/earth/ClimatePrediction/testing_suite_3.2.2_production/source[master]  > cat /esnas/autosubmit/t00q/status/t00q_20180207_1619.txt
t00q_LOCAL_SETUP COMPLETED /tmp/LOG_/t00q/t00q_LOCAL_SETUP.20180130133431.out /tmp/LOG_/t00q/t00q_LOCAL_SETUP.20180130133431.err

t00q_19890101_fc0_1_SIM FAILED  
t00q_19890101_fc0_2_SIM WAITING  
...
WS/shared/earth/ClimatePrediction/testing_suite_3.2.2_production/source[master]  > autosubmit --version
3.10.0
```

**Comment by kinow** [comment]
In GitLab by @pablohe on Feb 7, 2018, 16:21

reopened

**Comment by kinow** [comment]
In GitLab by @ lbatista on Feb 7, 2018, 16:44

@pablohe Sorry, my - typo - mistake for status FAILED. Tomorrow morning I will update the version that includes them as well

**Comment by kinow** [comment]
In GitLab by @ lbatista on Feb 8, 2018, 10:34

closed via commit f66d424b0b59459507bea06dbc860db82312e3b1

**Comment by kinow** [comment]
In GitLab by @ lbatista on Mar 6, 2018, 18:06

mentioned in commit f66d424b0b59459507bea06dbc860db82312e3b1
